src/scheduler.cc

```diff
@@ -296,12 +296,15 @@ Status SchedulerService::ExportFunction(ServerContext* context, const ExportFunc
   for (size_t i = 0; i < workers->size(); ++i) {
     ClientContext import_context;
     ImportFunctionRequest import_request;
-    import_request.mutable_function()->set_implementation(request->function().implementation());
+    import_request.mutable_function()->CopyFrom(request->function());
     if ((*workers)[i].current_task != ROOT_OPERATION) {
       ImportFunctionReply import_reply;
       (*workers)[i].worker_stub->ImportFunction(&import_context, import_request, &import_reply);
     }
   }
+  // TODO(rkn): Does this do a deep copy?
+  std::unique_ptr<Function> function(new Function(request->function()));
+  exported_functions.emplace_back(std::move(function));
```

Comment:
What is not safe is allowing the possibility of anything to happen between the call to `new` and the call that takes ownership of the pointer (which can happen if you have multiple expressions being evaluated independently on a line). An easy way to prevent that is to always construct `unique_ptr` on its own line, but that's not necessary and it's not related to `vector`.